static/js/gamedata/mapLoader.js

```diff
@@ -70,10 +70,61 @@ export function randomMap (sizeX, sizeY) {
   return Array.from(Array(sizeX * sizeY)).map(() => intToBase64style(fieldToNum(generateRandomField()))).join('')
 }
 
+export function compressedStringToMap (strCompressed) {
+  // Inverse to mapToString.
+  return decompress(strCompressed).then(stringToMap);
+}
+
 export function stringToMap (str) {
-  return {sizeX: parseInt(str.substr(0, 2)), sizeY: parseInt(str.substr(2, 2)), data: (str.substr(4).match(/.{1,3}/g) || []).map(field => numToField(base64StyleToInt(field)))}
+  return Promise.resolve({sizeX: parseInt(str.substr(0, 2)), sizeY: parseInt(str.substr(2, 2)), data: (str.substr(4).match(/.{1,3}/g) || []).map(field => numToField(base64StyleToInt(field)))})
```

Comment:
Returning a `Promise` here is not strictly necessary, though I think it's helpful for us to have `compressedStringToMap` an `stringToMap` both return the same type.

Reply:
I would make this an `async function` and then return normally instead of returning Promise.resolve. Makes it easy to see that it returns a Promise just by checking the signature.

Reply:
That's a great suggestion, done.